subprojects/core/src/integTest/groovy/org/gradle/initialization/CalculateTaskGraphBuildOperationIntegrationTest.groovy

```diff
@@ -122,10 +144,184 @@ class CalculateTaskGraphBuildOperationIntegrationTest extends AbstractIntegratio
+            task.buildPath == [":", ":", ":", ":", ":", ":", ":", ":"]
+            dependencies.taskPath.collect { it.sort() } == [[":compileJava"], [], [":compileJava", ":processResources"], [], [], [], [":anotherTask", ":otherTask"], []]
+            dependencies.buildPath == [[":included-build"], [], [":", ":"], [], [], [], [":", ":"], []]
+            finalizedBy.taskPath == [[], [], [], [], [], [], [":lastTask"], []]
+            mustRunAfter.taskPath == [[], [], [], [], [], [], [":firstTask"], []]
+            shouldRunAfter.taskPath == [[], [], [], [], [], [], [":secondTask"], []]
+        }
+        with(operations()[1].result.taskPlan) {
+            task.taskPath == [':compileJava']
+            task.buildPath == [':included-build']
+        }
+    }
+
+    @ToBeFixedForInstantExecution
+    def "exposes plan details with nested artifact transforms"() {
+        file('producer/src/main/java/artifact/transform/sample/producer/Producer.java') << """
+            package artifact.transform.sample.producer;
+
+            public final class Producer {
+
+                public String sayHello(String name) {
+                    return "Hello, " + name + "!";
+                }
+            }
+        """
+        file('producer/build.gradle') << """
+            apply plugin:'java-library'
+        """
+
+        buildFile << """
+            import org.gradle.api.artifacts.transform.TransformParameters
+            import org.gradle.api.artifacts.transform.TransformAction
+
+            plugins {
+                id 'java'
+                id 'application'
+            }
+
+            def artifactType = Attribute.of('artifactType', String)
+            def minified = Attribute.of('minified', Boolean)
+            def optimized = Attribute.of('optimized', Boolean)
+            dependencies {
+                attributesSchema {
+                    attribute(minified)
+                    attribute(optimized)
+                }
+                artifactTypes.getByName("jar") {
+                    attributes.attribute(minified, false)
+                }
+                artifactTypes.getByName("jar") {
+                    attributes.attribute(optimized, false)
+                }
+            }
+
+            configurations.all {
+                afterEvaluate {
+                    if (canBeResolved) {
+                        attributes.attribute(minified, true)
+                        attributes.attribute(optimized, true)
+                    }
+                }
+            }
+
+            dependencies {
+                registerTransform(SomeTransform) {
+                    from.attribute(minified, false).attribute(artifactType, "jar")
+                    to.attribute(minified, true).attribute(artifactType, "jar")
+                }
+                registerTransform(SomeTransform) {
```

Comment:
I don't think this will work, since Gradle can't determine the order of the transforms. Here is an example for a chained transform: https://github.com/gradle/gradle/blob/1eff263eb9b6c7e38c8ac15c31f945ddd2734288/subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformCachingIntegrationTest.groovy#L142-L243

Reply:
The test passes so it does seem to work.